packages/tx/test/eip7702.spec.ts

```diff
@@ -0,0 +1,201 @@
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: `0x${'20'.repeat(21)}`,
+              nonce: [],
+              yParity: '0x1',
+              r: ones32,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'address length should be 20 bytes')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: `0x${'20'.repeat(20)}`,
+              nonce: ['0x1', '0x2'],
+              yParity: '0x1',
+              r: ones32,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'nonce list should consist of at most 1 item')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: `0x${'20'.repeat(20)}`,
+              nonce: ['0x1'],
+              yParity: '0x1',
+              r: ones32,
+              s: undefined as never,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 's is not defined')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: `0x${'20'.repeat(20)}`,
+              nonce: ['0x1'],
+              yParity: '0x1',
+              r: undefined as never,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'r is not defined')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: `0x${'20'.repeat(20)}`,
+              nonce: ['0x1'],
+              yParity: undefined as never,
+              r: ones32,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'yParity is not defined')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: `0x${'20'.repeat(20)}`,
+              nonce: undefined as never,
+              yParity: '0x1',
+              r: ones32,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'nonce is not defined')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: undefined as never,
+              nonce: ['0x1'],
+              yParity: '0x1',
+              r: ones32,
+              s: ones32,
+            },
+          ],
```

Comment:
These repeated test data lists make this whole test setup pretty expansive. Can we clean this up by taking a base authorization list item data set (or the full authorization list, whatever fits better), and then only overwrite the data item to test (can maybe be done in one PR togehter with the interface movings to tx, see above).

Guess that's something to //cc @jochem-brouwer in.

Reply:
Yes, that would be a better approach in general, will address.